### PR TITLE
Simple markdown build does not touch tocmap

### PR DESCRIPTION
**Pull request by yufeih**
[AB#338814](https://dev.azure.com/ceapex/Engineering/_workitems/edit/338814/)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6823)

### CODE REVIEW

**Comment by yufeih** [comment]
The recent validation service deployment regressed `ApexValidationLibrary`, hence the diff.

**Comment by yufeih** [comment]
@irushx Output of `ms.author` is not stable

**Comment by yufeih** [comment]
Diffs expected

File: src/docfx/build/DocsetBuilder.cs [hunk 1]
```diff
@@ -141,7 +141,7 @@ public void Build(params string[] files)
 
                 var filesToBuild = files.Length > 0
                     ? files.Select(file => FilePath.Content(new PathString(file))).Where(file => _input.Exists(file) && _buildScope.Contains(file.Path)).ToHashSet()
-                    : publishUrlMap.GetAllFiles();
+                    : publishUrlMap.GetFiles().Concat(tocMap.GetFiles()).ToHashSet();
```
Comment (live1206) [inline src/docfx/build/DocsetBuilder.cs:144]: `ToHashSet` is not needed anymore?

File: src/docfx/publish/PublishUrlMap.cs [hunk 2]
```diff
@@ -80,34 +80,41 @@ public IEnumerable<FilePath> GetFilesByUrl(string url)
...
-                    () => ParallelUtility.ForEach(
-                        _errors, _redirectionProvider.Files.Where(x => x.Origin != FileOrigin.Fallback), file => AddItem(builder, file)),
-                    () => ParallelUtility.ForEach(
-                        _errors,
-                        _buildScope.GetFiles(ContentType.Resource).Where(x => x.Origin != FileOrigin.Fallback || _config.OutputType == OutputType.Html),
-                        file => AddItem(builder, file)),
-                    () => ParallelUtility.ForEach(
-                        _errors, _buildScope.GetFiles(ContentType.Page).Where(x => x.Origin != FileOrigin.Fallback), file => AddItem(builder, file)),
-                    () => ParallelUtility.ForEach(_errors, _tocMap.GetFiles(), file => AddItem(builder, file)));
+                return CreateUrlMap(
```
Comment (live1206) [inline src/docfx/publish/PublishUrlMap.cs:87]: After this change, `toc` is not part of `urlMap` anymore?

Comment (live1206) [reply src/docfx/publish/PublishUrlMap.cs:87]: Right now, toc is not using `urlMap`. It might use in the future.

**Review by live1206 — APPROVED** [review]

File: src/docfx/publish/PublishUrlMap.cs (hunk 2 above)
Comment (yufeih) [reply src/docfx/publish/PublishUrlMap.cs:87]: yep.

**Pull request merged**